File: AdegaOak/AdegaOak.Models/Models/Usuario.cs
namespace AdegaOak.Models.Models;

public class Usuario
{
    public int Id { get; set; }
    public string Nome { get; set; } = string.Empty;
    public string Username { get; set; } = string.Empty;
    public string PasswordHash { get; set; } = string.Empty;
    public string Role { get; set; } = "funcionario"; // "admin" | "funcionario"
    public bool Ativo { get; set; } = true;
    public DateTime CriadoEm { get; set; } = DateTime.UtcNow;

    // Navigation
    public ICollection<Movimentacao> Movimentacoes { get; set; } = [];
    public ICollection<ComboVenda> ComboVendas { get; set; } = [];
}

File: AdegaOak/frontend/types/usuario.ts
export interface UsuarioDto {
  id: number;
  nome: string;
  username: string;
  role: 'admin' | 'funcionario';
  ativo: boolean;
  criadoEm: string;
}

export interface CreateUsuarioRequest {
  nome: string;
  username: string;
  password: string;
  role: 'admin' | 'funcionario';
}

export interface UpdateUsuarioRequest {
  nome?: string;
  password?: string;
  role?: 'admin' | 'funcionario';
  ativo?: boolean;
}

export interface LoginRequest {
  username: string;
  password: string;
}

export interface LoginResponse {
  token: string;
  username: string;
  nome: string;
  role: string;
}
